File: trunk/ShadowFrameworkLite20/src/shadow/system/data/objects/SFVertex3fData.java
package shadow.system.data.objects;

import shadow.math.SFVertex3f;

public class SFVertex3fData extends SFVectorData{
	
	private SFVertex3f vertex;
	
	public SFVertex3fData(float x,float y,float z) {
		super(3);
		vertex=new SFVertex3f();
		vertex.setArray(getFloatValues());
		vertex.set3f(x,y,z);
	}
	
	public SFVertex3fData() {
		super(3);
		vertex=new SFVertex3f();
		vertex.setArray(getFloatValues());
	}

	public SFVertex3f getVertex3f(){
		return vertex;
	}
	
	@Override
	public SFVectorData clone() {
		return new SFVertex3fData();
	}
}

File: SFDataApplicationTestC++/shadow/system/data/objects/SFVertex3fData.cpp

#include "SFVertex3fData.h"


namespace sf{

SFVertex3fData::SFVertex3fData(float x,float y,float z) :SFVectorData(vertex.getV(),3){
    vertex.set3f(x,y,z);
}

SFVertex3fData::SFVertex3fData() :SFVectorData(vertex.getV(),3){
	vertex.set3f(0,0,0);
}

SFVertex3f SFVertex3fData::getVertex3f(){
	return vertex;
}

SFVectorData* SFVertex3fData::clone() {
	return new SFVertex3fData();
}
}
